app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,33 @@
+class FriendshipsController < ApplicationController
+  before_action :require_user
+  def index;end
+
+  def create
+    leader = User.find(params[:leader_id])
+    if current_user.id == params[:leader_id].to_i
```

Comment:
Consider that you already have `leader` object stored under variable.

`if current_user == leader` is much easier.